text/0046-metadata-for-offline-signers.md

```diff
@@ -264,7 +368,9 @@ Some slightly out of spec systems might experience breaking changes as new conte
 
 ## Testing, Security, and Privacy
 
-The metadata shortening protocol should be extensively tested on all available examples of metadata before releasing changes to either metadata or shortener. Careful code review should be performed on shortener implementation code to ensure security. The main metadata tree would inevitably be constructed on runtime build which would also ensure correctness.
+All implementations are required to follow strictly the RFC to generate the metadata hash. This includes which hash function to use and how to construct the metadata types tree. So, all implementations are following the same security criteria. As the chains will calculate the metadata hash at compile time, the build process needs to be trusted. However, this is already a solved problem in the Polkadot ecosystem by using reproducible builds. So, anyone can rebuild a chain runtime to ensure that a proposal is actually containing the changes as advertised.
```

Comment:
```suggestion
All implementations MUST follow the RFC to generate the metadata hash. This includes which hash function to use and how to construct the metadata types tree. So, all implementations are following the same security criteria. As the chains will calculate the metadata hash at compile time, the build process MUST be trusted. However, this is already a solved problem in the Polkadot ecosystem using reproducible builds. So, anyone can rebuild a chain runtime to ensure that a proposal is actually containing the changes as advertised.
```

Reply:
the second MUST - if it's assumed that actors re-build it, it's not trust, is it?

Reply:
Hmm, should we use SHALL?

Reply:
idk, I just copy-pasted proposed wording from reviewers verbatim here. I don't think we need a control verb here, as we are not requiring that part. The build process is governed elsewhere and we only follow design outlined there.

Actually, I'm not sure we should talk about trusting build process here at all, but I also see that it clarifies the design a bit. Yet, if someone builds a runtime using different build system (or even uses another implementation altogether), the ecosystem would still be protected by all the other nodes of given chain requiring agreement on digest value. So what's the point of trusting build process then? We do seem to have some reliance on single build shop, but that's not really something to be proud of.